### PR TITLE
Cordova

### PR DESCRIPTION
**Pull request by StephenCoady**
Creating this PR so it is easier to see what will change if we go with Cordova.

To run this: 

clone and link the javascript sdk
from within this repo:
```
cd app
npm i
npm run watch-start
```

### CODE REVIEW

**Comment by darahayes** [comment]
@StephenCoady I believe it's standard practice to put cordova plugins in `.gitignore`

File: README.md [hunk 1]
```diff
@@ -1,22 +1,82 @@
...
 
 Investigation is using Apollo link to perform filtering for graphQL network layer.
 We have build simple React Starter App that adds new users.
 We can simulate conflicts by editing data on the server and making change when offline.
 
+### Pre-requisites
+
+cordova:
+```
+npm i -g cordova@8.1.2
```
Comment (wtrocki) [inline README.md:27]: [Optional] [FYI] best practice was to avoid global and install it in package.

File: server-apollo/src/schema.js [hunk 2]
```diff
@@ -34,7 +34,7 @@ type Subscription {
 const resolvers = {
   Query: {
     allUsers: async (obj, args, context) => {
-      const result = context.db.select().from('users')
+      const result = await context.db.select().from('users')
```
Comment (wtrocki) [inline server-apollo/src/schema.js]: this change will cause: TypeError: result.limit is not a function",

File: server-apollo/src/schema.js [hunk 3]
```diff
@@ -1,11 +1,12 @@
 const { gql } = require('apollo-server')
-const { makeExecutableSchema } = require('graphql-tools')
+const { makeExecutableSchema, SchemaDirectiveVisitor } = require('graphql-tools')
 const { GraphQLNonNull } = require('graphql')
 const { combineResolvers, pipeResolvers } = require('graphql-resolvers')
 const { pubSub, EVENTS } = require('./subscriptions')
 const { withConflict } = require('./sdk/withConflict')
 
 const typeDefs = gql`
+directive @onlineOnly on FIELD
```
Comment (wtrocki) [inline server-apollo/src/schema.js:9]: I guess this is an spike and it's not going to be merged right?
Directives should be added separately in framework itself.

Generally speaking I will strongly recommend to not do onlineOnly/skipSquash until we will get core offline functionality done. I believe that we have this done in sdk but there will be couple implications and complexity involved.

 OnlineOnly introduces so many corner cases (how to handle errors when this is done when offline etc.) that will make our changes so large and basically untestable in single chunk of work.

Let's  chat about it on monday but basically we can create separate tickets to handle those cases and do that after basic functionality is done and tested. 

+100 to land sdk changes as they make sense but there will be some testing and framework implications on the server that best to be done in separate PR's

Comment (StephenCoady) [reply server-apollo/src/schema.js:9]: After syncing up I agree that these two directives introduce too many edge cases that we just cannot test as part of this initial queue body of work. Will remove them from the app and just leave them as part of the sdk for now.

**Comment by wtrocki** [comment]
Closing as this is put on hold. Leaving branch for future reference.
